YahooFantasy.mjs

```diff
@@ -48,34 +48,51 @@ class YahooFantasy {
   }
 
   api(...args) {
-    const method = args.shift();
-    const url = args.shift();
-    const cb = args.pop();
-    let postData = false;
+    var options = {};
+    let cb = () => {};
+    let responseMapper = x => x;
+    if (typeof args[0] === "object") {
+      const argOptions = args.shift();
+      options.method = argOptions.method;
+      options.url = argOptions.url;
+      responseMapper = argOptions.responseMapper;
+      cb = args.pop() || cb;
+    } else {
+      options = {};
+      options.method = args.shift();
+      options.url = args.shift();
+  
+      if (args.length) {
+        if (typeof args[args.length - 1] === "function") {
+          cb = args.pop();
+        }
 
-    if (args.length) {
-      postData = args.pop();
+        if (args.length) {
+          options.body = args.pop();
+        }
+      }
     }
 
-    var options = {
-      url: url,
-      method: method,
-      json: true,
-      auth: {
-        bearer: this.yahooUserToken
-      }
-    };
-
-    request(options, (e, body, data) => {
-      if (e) {
-        return cb(e);
-      } else {
-        if (data.error) {
-          return cb(data.error);
-        }
+    if (!options.body) {
+      options.json = true;
+    }
 
-        return cb(null, data);
-      }
+    options.auth = { bearer: this.yahooUserToken };
+    return new Promise((resolve, reject) => {
```

Comment:
so... if i'm understanding properly...

you return a Promise regardless, it just always happens.
you make the request to the API and then resolve/reject the promise and call the callback.

I've just never seen this before... is this a standard practice for things like this?